demos/common/python/models/hpe_hrnet.py

```diff
@@ -0,0 +1,136 @@
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+
+from .hpe_associative_embedding import AssociativeEmbeddingDecoder
+from .model import Model
+from .utils import resize_image
+
+
+class HpeHRNet(Model):
+    def __init__(self, ie, model_path, target_size, aspect_ratio, prob_threshold, size_divisor=32):
+        super().__init__(ie, model_path)
+        self.image_blob_name = self._get_inputs(self.net)
+        self.heatmaps_blob_name = find_layer_by_name('heatmaps', self.net.outputs)
+        try:
+            self.nms_heatmaps_blob_name = find_layer_by_name('nms_heatmaps', self.net.outputs)
+        except ValueError:
+            self.nms_heatmaps_blob_name = self.heatmaps_blob_name
+        self.embeddings_blob_name = find_layer_by_name('embeddings', self.net.outputs)
+
+        self.output_scale = self.net.input_info[self.image_blob_name].input_data.shape[-1] / self.net.outputs[self.heatmaps_blob_name].shape[-1]
+
+        if target_size is None:
+            h, w = self.net.input_info[self.image_blob_name].input_data.shape[-2:]
+            target_size = min(h, w)
+        if aspect_ratio >= 1.0:  # img width >= height
+            input_height, input_width = target_size, round(target_size * aspect_ratio)
+        else:
+            input_height, input_width = round(target_size / aspect_ratio), target_size
+        self.h = (input_height + size_divisor - 1) // size_divisor * size_divisor
+        self.w = (input_width + size_divisor - 1) // size_divisor * size_divisor
+        default_input_shape = self.net.input_info[self.image_blob_name].input_data.shape
+        input_shape = {self.image_blob_name: (default_input_shape[:-2] + [self.h, self.w])}
+        self.logger.info('Reshape net to {}'.format(input_shape))
+        self.net.reshape(input_shape)
+
+        self.decoder = AssociativeEmbeddingDecoder(
+            num_joints=self.net.outputs[self.heatmaps_blob_name].shape[1],
+            adjust=True,
+            refine=True,
+            delta=0.5,
+            max_num_people=30,
+            detection_threshold=0.1,
+            tag_threshold=1,
+            pose_threshold=prob_threshold,
+            use_detection_val=True,
+            ignore_too_much=False,
+            dist_reweight=True)
+        self.size_divisor = size_divisor
+
+    @staticmethod
+    def _get_inputs(net):
+        image_blob_name = None
+        for blob_name, blob in net.input_info.items():
+            if len(blob.input_data.shape) == 4:
+                image_blob_name = blob_name
+            else:
+                raise RuntimeError('Unsupported {}D input layer "{}". Only 4D input layers are supported'
```

Comment:
Could you remove `2D` for `HpeAssociativeEmbedding` as well

Reply:
Removed